code/go/0chain.net/blobbercore/filestore/state.go

```diff
@@ -0,0 +1,210 @@
+package filestore
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/datastore"
+	"github.com/0chain/blobber/code/go/0chain.net/core/common"
+	"github.com/0chain/blobber/code/go/0chain.net/core/logging"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+)
+
+type allocation struct {
+	mu            *sync.Mutex
+	allocatedSize uint64
+	filesNumber   uint64
+	filesSize     uint64
+
+	tmpMU       *sync.Mutex
+	tmpFileSize uint64
+}
+
+func (fs *FileStore) setAllocation(allocID string, alloc *allocation) {
+	fs.rwMU.Lock()
+	fs.mAllocs[allocID] = alloc
+	fs.rwMU.Unlock()
+}
+
+func (fs *FileStore) getAllocation(allocID string) *allocation {
+	fs.rwMU.RLock()
+	alloc := fs.mAllocs[allocID]
+	fs.rwMU.RUnlock()
+	return alloc
+}
+
+func (fs *FileStore) removeAllocation(ID string) {
+	fs.rwMU.Lock()
+	delete(fs.mAllocs, ID)
+	fs.rwMU.Unlock()
+}
+
+func (fs *FileStore) initMap() error {
+	ctx, cnCl := context.WithCancel(context.Background())
+	defer cnCl()
+
+	ctx = datastore.GetStore().CreateTransaction(ctx)
+	db := datastore.GetStore().GetTransaction(ctx)
+	if db == nil {
+		return errors.New("could not get db client")
+	}
+
+	limitCh := make(chan struct{}, 50)
+	wg := &sync.WaitGroup{}
+	var allocations []*dbAllocation
+
+	err := db.Model(&dbAllocation{}).FindInBatches(&allocations, 1000, func(tx *gorm.DB, batch int) error {
+		for _, alloc := range allocations {
+			a := allocation{
+				allocatedSize: uint64(alloc.BlobberSize),
+				mu:            &sync.Mutex{},
+				tmpMU:         &sync.Mutex{},
+			}
+
+			fs.setAllocation(alloc.ID, &a)
```

Comment:
We can add a new method say `fs.setAllocations()` to add all allocations in one calling within one lock/unlock block.